src/cohort/Cohort.ts

```diff
@@ -0,0 +1,41 @@
+import Student from "../student/Student";
+import StudentSearchOptions from "../student/StudentSearchOptions";
+import ManagerUtils from "../utils/ManagerUtils";
+
+export default class Cohort {
+  name: string;
+  students: number[]; // Stores IDs instead of instances
+  capacity = 24;
+
+  constructor(name: string, students?: number[]) {
+    this.name = name;
+    this.students = students ?? [];
+  }
+
+  addStudent(options: StudentSearchOptions) : boolean {
+    if(this.students.length >= this.capacity) throw new Error("Cohort is full");
+    const student = ManagerUtils.findStudent(options) ;
+    if(student === undefined) throw new Error("Student doesn't exist");
+    if(ManagerUtils.findStudentCohort(options) !== undefined) throw new Error("Student already in another cohort");
+    if (this.students.includes(student.id))
+      throw new Error("Student already exists in cohort");
+    else this.students.push(student.id);
+    return true;
+  }
+
+  getStudents() : Student[] {
```

Comment:
This is unknown to me, is it TypeScript syntax?

Reply:
Yes ! It makes everything easier to work with since all objects are typed and we can know what we're working with without having to guess properties